tests/readmem.rs

```diff
@@ -33,7 +34,7 @@ fn read_memory() -> Result<(), Box<dyn std::error::Error>> {
 
     assert_eq!(&rval, b"Hello, world!");
 
-    target.unpause()?;
+    target.unpause(&target)?;
```

Comment:
Hi @nbaksalyar 
I don't think this is an Idiomatic API by design, and its mostly cause of the weird structure in the `UnixTarget` trait.
Could you please provide your inputs here.

Reply:
Trais in Rust don't have properties yet, but we can simply work around by having an abstract function like this:

```rust
trait UnixTarget {
  fn pid(&self) -> Pid;
}
```

Then it can be implemented e.g. for the Linux target struct:

```rust
impl LinuxTarget {
  fn pid(&self) -> Pid { self.pid }
}
```

and then in `unpause` it should suffice to call `self.pid()` to obtain the process ID.
And since all *nixes have the concept of PID, it's a good idea to introduce this getter function anyway!

Reply:
Done